src/js/adc.js

```diff
@@ -13,14 +13,16 @@
  * limitations under the License.
  */
 
-function Adc() {
-  if (!(this instanceof Adc)) {
-    return new Adc();
-  }
-}
-
-Adc.prototype.open = function(configuration, callback) {
-  return new native.Adc(configuration, callback);
+var adc = {
+  open: function(config, callback) {
+    var adcPin = new native(config, function(err) {
+      callback(err, adcPin);
```

Comment:
@bzsolt What happens when user calls `open` without `callback` parameter?

Reply:
It will throw a TypeError.

Reply:
@bzsolt Although the behavior is changed from previous one, it seems reasonable. Also, I confirmed node also throws `TypeError`.

```
$ node -e "require('fs').open('/etc/hosts','r')"
fs.js:131
    throw new TypeError('"callback" argument must be a function');
    ^
```